server/src/main/java/org/opensearch/common/TransferPartStreamSupplier.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common;
+
+import java.io.IOException;
+
+/**
+ * TransferPartStreamSupplier is used to supply streams to specific parts of a file based on
+ * the partNo, size and position (the offset in the file)
+ */
+public interface TransferPartStreamSupplier {
```

Comment:
Should this be a FunctionalInterface?

Reply:
We do need a new interface since we need to be able to throw exceptions from the supply method, but we can generalize it to something like a `CheckedTriFunction`, similar to https://github.com/opensearch-project/OpenSearch/blob/c659d04956e300740a7626f952bb7958492ed843/server/src/main/java/org/opensearch/common/CheckedBiFunction.java#L41